staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -333,25 +337,45 @@ func ValidateWebhookConfiguration(fldPath *field.Path, c *api.WebhookConfigurati
 	default:
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("connectionInfo", "type"), c.ConnectionInfo, []string{"InClusterConfig", "KubeConfigFile"}))
 	}
+	if len(c.MatchConditions) > 64 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("matchConditions"), c.MatchConditions, "must not exceed 64 matchConditions"))
+	}
+	for i, condition := range c.MatchConditions {
+		fldPath := fldPath.Child("matchConditions").Index(i).Child("expression")
+		if len(strings.TrimSpace(condition.Expression)) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath, ""))
+		} else {
+			allErrs = append(allErrs, validateMatchConditionsExpression(fldPath, condition.Expression)...)
+		}
+	}
+	return allErrs
+}
 
-	// TODO: Remove this check and ensure that correct validations below for MatchConditions are added
-	// for i, condition := range c.MatchConditions {
-	//	 fldPath := fldPath.Child("matchConditions").Index(i).Child("expression")
-	//	 if len(strings.TrimSpace(condition.Expression)) == 0 {
-	//	     allErrs = append(allErrs, field.Required(fldPath, ""))
-	//	 } else {
-	//		 allErrs = append(allErrs, ValidateWebhookMatchCondition(fldPath, sampleSAR, condition.Expression)...)
-	//	 }
-	// }
-	if len(c.MatchConditions) != 0 {
-		allErrs = append(allErrs, field.NotSupported(fldPath.Child("matchConditions"), c.MatchConditions, []string{}))
+func validateMatchConditionsExpression(fldPath *field.Path, expression string) field.ErrorList {
```

Comment:
I am not sure if you wanna use the same validation machinist here or if the scope is the same but here is for reference: https://github.com/kubernetes/kubernetes/blob/d953755686d64207740c15c7eb1599f874f29243/pkg/apis/admissionregistration/validation/validation.go#L1053